src/api/gamestate.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "types.h"
+
+namespace api {
+
+mahjong::GameSettings convertGameSettings(const CGameSettings* settings);
+
+extern "C" {
+
+int StartGame(const CGameSettings* settings, int async);
+void ExitGame(int game);
+mahjong::GameState* InitGameState(const CGameSettings* settings);
+mahjong::GameState* AdvanceGameState(mahjong::GameState* state);
+void DestroyGameState(mahjong::GameState* state);
```

Comment:
For C api layers this would more normally be called FreeGameState rather than destroy. Or FreeGameStateObj if you prefer, but then you might want to update the other names for consistency. Up to you.